### PR TITLE
Rails style rule for route actions order

### PR DESCRIPTION
**Pull request by pfac**
Why:
- We have no rule for the order to follow when stating the routes to be
  enabled with the `:only` option, which decreases the consistency of
  the code in the routes file.

This change addresses the need by:
- Proposing the resource lifecycle as the preferred order for stating
  routes.

### CODE REVIEW

**Comment by zamith** [comment]
What about `edit` and `update`? I don't think there is a clear cut order, at least not based on "lifecycle". We could potentially define an order, and just not sure this is enough of a pain to worry about it this much.

To be honest, I never had any issue understanding this as I usually will use `rake routes` when I'm unsure if a route exists. What was the issue that triggered this?

**Comment by pfac** [comment]
> What about `edit` and `update`?

Those are only missing in the example, because if we were using all routes we wouldn't use the `:only` option. The complete list is stated in the style rulle itself.

> I don't think there is a clear cut order, at least not based on "lifecycle".

I see your point. I called it lifecycle because that's the normal flow a user will visit the routes:
1. Open the `index` page, which will initially be empty but will list existing resources;
2. Go to the `new` page to add a new resource;
3. Submit form to `create`;
4. Be redirected to or visit `show` page to see resource's page;
5. Go to the `edit` page to modify the resource;
6. Submit form to `update`;
7. `destroy` the resource.

> We could potentially define an order, and just not sure this is enough of a pain to worry about it this much.

I don't see why we need something to be a big pain to have a style convention about it. Plenty of stuff in these guides were never a pain (as far as I know), doesn't make them less valid.

> To be honest, I never had any issue understanding this as I usually will use `rake routes` when I'm unsure if a route exists. What was the issue that triggered this?

I didn't understand this last part. This is just the result of a discussion me and @gabrielpoca were having in the project we're working on. I prefer not to leave it open, and have a rule that dictates how it should be done, instead of doing it adhoc. I believe it makes routes easier to read and reason about.

Still, it's my opinion. Feel free to disagree.

**Comment by zamith** [comment]
I meant that I don't think it matters enough to warrant an order. I believe it's very different from css rules, which we write a lot more of and span multiple lines, and even for those we weren't comfortable having an order before we had a linter to check it for us.

Here the situation is similar in that sense, in my opinion, the order is not super clear (for instance, rake routes uses the order `index`, `create`, `new`, `edit`, `show`, `update`, `destroy`) and thus I wouldn't like to have it added to the guides unless there is a linter checking whatever order we come up with.

That being said, if for instance rubocop is checking it, I don't have a strong enough feeling against this to refute it.

**Comment by naps62** [comment]
TL;DR: I'm with zamith on this

I never felt this to be an issue, or to decrease readability.
The `rake routes | fzf` output doesn't seem to care about this order,

> I don't see why we need something to be a big pain to have a style convention about it. Plenty of stuff in these guides were never a pain (as far as I know), doesn't make them less valid.

This is the part I disagree the most
Adding rules _just because_ will only serve to increase the size of the guides

This seemed to originate from a discussion on a particular project, and it can certainly be applied there if it's developers agree.
But I think expanding it to a company-wide rule seems a bit overkill

**Comment by gabrielpoca** [comment]
Another argument in the code review that originated this was that the order makes the file look like it was written by the same person.

I'm also with Miguel and Zamith on this; I would rather not care about the order. I could agree if Rubocop caught it, but I don't see the value.

**Comment by pfac** [comment]
Ok, I'm closing this PR since the majority does not agree with this change.
